### PR TITLE
Restore Target Size (Enhanced) Understanding Doc content

### PR DESCRIPTION
**Pull request by giacomo-petri**
Closes: #3617

This PR restores content in the 2.5.5 Target Size (Enhanced) understanding document that was lost during the transition from 2.1 (when it was just "Target Size") to 2.2 (after the 2.5.8 Minimum version was added, and 2.5.5 become "Enhanced").
Specifically, it adds back:
- exceptions in the intent section
- additional examples

Some code indentations were also modified to improve code readability. These do not affect the resulting HTML document.

The resulting Understanding document (without CSS styling) can be viewed [here](https://raw.githack.com/w3c/wcag/giacomo-petri-target-size-enhanced-patch/understanding/21/target-size-enhanced.html)

### CODE REVIEW

**Comment by giacomo-petri** [comment]
To be discussed:

`<li>Ensuring inline links provide sufficiently large activation target.</li>`

This technique regarding the adequate activation target for inline links has been relocated from Advisory Technique in 2.1 to Sufficient Technique in 2.2. It's noteworthy that inline links constitute an exception for the particular Success Criterion.

**Comment by patrickhlauke** [comment]
> To be discussed:
> 
> `<li>Ensuring inline links provide sufficiently large activation target.</li>`
> 
> This technique regarding the adequate activation target for inline links has been relocated from Advisory Technique in 2.1 to Sufficient Technique in 2.2. It's noteworthy that inline links constitute an exception for the particular Success Criterion.
> 
> While the other modifications were seemingly dropped inadvertently with the transition from 2.1 to 2.2, it appears that this particular change was a deliberate decision. We must now determine our intentions regarding it.

the technique should still remain an advisory technique, as it goes beyond what 2.5.5 (and 2.5.8) demand

**Comment by giacomo-petri** [comment]
I totally agree with you @patrickhlauke.
I've edited the file reflecting this choice.

**Review by patrickhlauke — APPROVED** [review]

**Comment by bruce-usab** [comment]
See GitHack [preview of page (no CSS)](https://raw.githack.com/w3c/wcag/giacomo-petri-target-size-enhanced-patch/understanding/21/target-size-enhanced.html) since the raw html is harder to read.

**Review by bruce-usab — CHANGES_REQUESTED** [review]
Thanks @giacomo-petri for writing up.

One questions I have, Example 3 Anchor Link :
> The target is an in-page link and the target is less than 44 by 44 CSS pixels. _Users can scroll the screen using browser functions so target size does not need to be met._

I do not agree that the ability to scroll is justification for small targets.  I am open to be swayed.  I think hypertext typically is excepted by being part of a sentence.  If the target is part of in-page menu, would it not need to meet target size?

**Comment by giacomo-petri** [comment]
Hi @bruce-usab ,

I haven't made any edits to this section of the Understanding document. The purpose of this pull request was to bring the 2.1 Understanding document in line with the 2.2 Understanding document, as the latter was incomplete with several sections missing compared to the former.

The efficiency of using "scroll" as an alternative to the anchor link can be subject to debate, despite providing a means to access the content. However, as you've pointed out, this doesn't constitute an exception within the normative document, so it should be considered non-compliant.

While it wasn't the intention of the pull request, I'm open to removing the mentioned example if you agree with that course of action.

**Review by bruce-usab — APPROVED** [review]

**Comment by bruce-usab** [comment]
> I haven't made any edits to this section of the Understanding document. 

Thanks @giacomo-petri !  That is okay then, my apologies.  Could you be prepared to walk the group through your changes on the call _next_ Friday, 8 March?

**Comment by bruce-usab** [comment]
@johannesFischer84 thanks for flagging the discrepancy.  Does this PR look good to you?

**Comment by johannesFischer84** [comment]
@bruce-usab / @giacomo-petri 
Thank your for correcting this quickly. It looks great!

**Comment by bruce-usab** [comment]
Discussed on TF call 4/12 and Ready for approval.

File: understanding/21/target-size-enhanced.html [hunk 1]
```diff
@@ -7,87 +7,103 @@
...
+      <p>The issue can be further complicated for responsive/mobile sites which need to accommodate different types of fine and coarse inputs (e.g. a site that can be accessed both on a traditional desktop/laptop with a mouse, as well as on a tablet or mobile phone with a touch screen).</p>
+      <p>While this criterion defines a minimum target size, it is recommended that larger sizes are used to reduce the possibility of unintentional actions. This is particularly relevant if any of the following are true:</p>
+  	  <ul>
+        <li>the control is used frequently;</li>
+        <li>the result of the interaction cannot be easily undone;</li>
+        <li>the control is positioned where it will be difficult to reach, or is near the edge of the screen;</li>
+        <li>the control is part of a sequential task.</li>
+  	  </ul>
+      <p>The targets on a screen can have different purposes and uses, and this Success Criterion specifies how each is to be handled.</p>
```
Comment (detlevhfischer) [inline understanding/21/target-size-enhanced.html:30]: > and this Success Criterion specifies how each is to be handled.

This sounds decidedly odd. The understanding doc might but the normative text does not specify this in great detail (for good reasons).

File: understanding/21/target-size-enhanced.html [hunk 2]
```diff
@@ -7,87 +7,103 @@
...
+      <p><strong>Equivalent targets:</strong> If there is more than one target on a screen that performs the same action, only one of the targets need to meet the target size of 44 by 44 CSS pixels.</p>
+      <p><strong>Inline:</strong> Content displayed can often be reflowed based on the screen width available. This is known as responsive design and makes it easier to read since you do not need to scroll both horizontally and vertically. In reflowed content, the targets can appear anywhere on a line and can change position based on the width of the available screen. Since targets can appear anywhere on the line, the size cannot be larger than the available text and spacing between the sentences or paragraphs, otherwise the targets could overlap. It is for this reason targets which are contained within one or more sentences are excluded from the target size requirements.</p>
```
Comment (detlevhfischer) [inline understanding/21/target-size-enhanced.html:32]: > can often be reflowed

I believe the better Einglish would be:
>can often reflow

or
> is often designed to reflow

**Review by detlevhfischer — CHANGES_REQUESTED** [review]
Two editorial suggestions. Left comments between lines.

**Comment by detlevhfischer** [comment]
[Rendered view of changes for convenience](https://raw.githack.com/w3c/wcag/e581deae5842084f9e17e37a0cd06aae685174e0/understanding/21/target-size-enhanced.html)

File: understanding/21/target-size-enhanced.html [hunk 3]
```diff
@@ -7,87 +7,103 @@
...
-            </section>
-</section>
-</body>
+    </section>
+  	<section id="examples">
+  	  <h2>Examples</h2>
+      <ul>      
+        <li><strong>Example 1: Buttons</strong><br> Three buttons are on-screen and the touch target area of each button is 44 by 44 CSS pixels.</li>
+        <li><strong>Example 2: Equivalent target</strong><br> Multiple targets are provided on the page that perform the same function. One of the targets is 44 by 44 CSS pixels. The other targets do not have a minimum touch target of 44 by 44 CSS pixels.</li>
+        <li><strong>Example 3: Anchor Link</strong><br> The target is an in-page link and the target is less than 44 by 44 CSS pixels. Users can scroll the screen using browser functions so target size does not need to be met.</li>
```
Comment (detlevhfischer) [inline understanding/21/target-size-enhanced.html:61]: > Users can scroll the screen using browser functions so target size does not need to be met.

That users can scroll to in-page links seems orthogonal to the case covered by the example. For inline linkstarget size is exempted regardless of whether scrolling is necessary. This sentence should be removed.

File: understanding/21/target-size-enhanced.html [hunk 4]
```diff
@@ -7,87 +7,103 @@
...
-		      </ul>
-            </section>
-</section>
-</body>
+    </section>
+  	<section id="examples">
+  	  <h2>Examples</h2>
+      <ul>      
+        <li><strong>Example 1: Buttons</strong><br> Three buttons are on-screen and the touch target area of each button is 44 by 44 CSS pixels.</li>
+        <li><strong>Example 2: Equivalent target</strong><br> Multiple targets are provided on the page that perform the same function. One of the targets is 44 by 44 CSS pixels. The other targets do not have a minimum touch target of 44 by 44 CSS pixels.</li>
```
Comment (detlevhfischer) [inline understanding/21/target-size-enhanced.html:60]: > The other targets do not have a minimum touch target of 44 by 44 CSS pixels.

This sentence seems to dangle, "the other targets" is unclear because it is not scoped to the target that meets the requirement because it has a sibling that does.

**Review by detlevhfischer — COMMENTED** [review]
I think the example texts need some work (see comments).

**Comment by giacomo-petri** [comment]
Hi @detlevhfischer,

this PR aims to align the Understanding Document of WCAG 2.2, which was unintentionally cut off, with the Understanding Document of WCAG 2.1.

Should we proceed with this and open a separate PR for further improvements?

**Comment by detlevhfischer** [comment]
Sure - I don’t mind if this becomes a separate PR…

File: understanding/21/target-size-enhanced.html (hunk 3 above)
Comment (johannesFischer84) [reply understanding/21/target-size-enhanced.html:61]: I prefer not to remove this sentence. It gives a kind of explanation why in-page links need not to be 44 x 44 px. But it is a possibility to use a different wording like for example this one:

> Anchor Link
The target is an in-page link and the target is less than 44 by 44 CSS pixels. Reason: No other page need to be opened. Users can get to the destination of the link by using the tab key or by scrolling the screen.

Comment (patrickhlauke) [reply understanding/21/target-size-enhanced.html:61]: I'd actually say this line is quite misleading. Yes, a user can scroll ... but that's provided they know/understand where that in-page link was linking to. That's not always a given.

**Comment by giacomo-petri** [comment]
Great, we can consider this as ready for approval.

@detlevhfischer, would you mind to open an issue regarding the particular edits you suggested?

Thanks

**Comment by bruce-usab** [comment]
Discussed on call 4/26 and left in For discussion.  It is problematic to have so much back-and-forth in a PR rather than an Issue.

File: understanding/21/target-size-enhanced.html (hunk 1 above)
Comment (mbgower) [reply understanding/21/target-size-enhanced.html:30]: It's important to point out that this is the pre-existing content for the 2.1 understanding document, being restored. I'm not disagreeing that it is awkwardly written in places, but the intent of this PR is to restore the prior version.

File: understanding/21/target-size-enhanced.html (hunk 4 above)
Comment (bruce-usab) [reply understanding/21/target-size-enhanced.html:60]: @detlevhfischer can you suggest phrasing that would add clarity?  I could not come up with anything pithy.

File: understanding/21/target-size-enhanced.html [hunk 5]
```diff
@@ -7,87 +7,103 @@
...
+        <p class="instructions">Although not required for conformance, the following additional techniques should be considered in order to make content more accessible. Not all techniques can be used or would be effective in all situations.</p>
+        <ul>
+            <li>Ensuring inline links provide sufficiently large activation target.</li>
+        </ul>
+      </section>
+      <section id="failure">
+        <h2>Failure</h2>
+        <p class="instructions">The following are common mistakes that are considered failures of Success Criterion 2.5.5 by the <abbr title="Web Content Accessibility Guidelines">WCAG</abbr> Working Group.</p>
+        <ul>
+          <li>Failure of success criterion 2.5.5 due to target being less than 44 by 44 CSS pixels.</li>
```
Comment (mbgower) [inline understanding/21/target-size-enhanced.html:104]:
```suggestion
          <li>Failure of Success Criterion 2.5.5 due to target being less than 44 by 44 CSS pixels.</li>
```

File: understanding/21/target-size-enhanced.html [hunk 6]
```diff
@@ -7,87 +7,103 @@
...
+  			<dt>Why it's important</dt><dd>Some people cannot tap small objects.</dd>
+  		</dl>
+  	</section>
+  	<section id="intent">
+  		<h2>Intent</h2>
+  		<p>The intent of this success criterion is to help users who may have trouble activating a small target because of hand tremors, limited dexterity or other reasons. If the target is too small, it may be difficult to aim at the target. Mice and similar pointing devices can be hard to use for these users, and a larger target will help them greatly in having positive outcomes on the web page.</p>
+      <p>Touch is particularly problematic as it is an input mechanism with coarse precision. Users lack the same level of fine control as on inputs such as a mouse or stylus. A finger is larger than a mouse pointer, and generally obstructs the user's view of the precise location on the screen that is being touched/activated.</p>
+      <p>The issue can be further complicated for responsive/mobile sites which need to accommodate different types of fine and coarse inputs (e.g. a site that can be accessed both on a traditional desktop/laptop with a mouse, as well as on a tablet or mobile phone with a touch screen).</p>
```
Comment (frankie-wolf) [inline understanding/21/target-size-enhanced.html:22]: Line 22: 
1. Replace the slashes with the word "or" to improve readability
2. Replace "which" with "that" (since it's followed by a restrictive clause)
3. Replace "e.g." with "for example" for plainer language
4. Consider rewriting this as two sentences to decrease the cognitive load

As one sentence:
"The issue can be further complicated for responsive and mobile sites that need to accommodate different types of fine and coarse inputs (for example, a site that can be accessed on a traditional desktop or laptop with a mouse, as well as on a tablet or mobile phone with a touch screen)."

Comment (mbgower) [reply understanding/21/target-size-enhanced.html:22]: Thanks, @frankie-wolf. 
The purpose of this PR was just to restore accidentally deleted material, not to alter that content.
There have been a number of suggestions by people of ways to improve the content, and our intention is to create a second PR which improves the information, not just restores it!
